### PR TITLE
config: indicate availability of FITS service

### LINKED ISSUE #59: disable FITS component/tab if FITS is not available (currently backend crashes)

### PR DESCRIPTION
**Pull request by larnhold**
#### What does this PR do?

Adds a new field to ConfigDO indicating if a non-empty URL "fits-url" has been defined in application.yaml

#### Breaking changes

No breaking changes

### Checks

- [ ] Tested with Oracle/PostgreSQL
- [ ] Export updated
- [ ] Documentation added
- [ ] Tests added
- [ ] E2e tests created
- [ ] Successfully ran e2e tests before merge

closes GH-59

### CODE REVIEW

**Review by ValentinFutterer — COMMENTED** [review]
I would suggest to check if the fits service is running and available, since currently it is only checked if the service has a url in the application.yaml. If the fits container crashes, the frontend still displays the fits component but trying to use it causes an error.

**Comment by rekt-hard** [comment]
The same can be said about all other services the back end interacts with. It might be that a service is available one second and unavailable the next second. We would have to monitor a services availability, update the configs accordingly and let the front end know (or periodically fetch the current status from the front end). Being this dynamic and reactive requires thorough changes in the front end and back end.

For now, I think it is a fair assumption that if a URL is specified, a service should be available in general.

File: src/main/java/at/ac/tuwien/damap/rest/ConfigResource.java [hunk 1]
```diff
@@ -44,7 +50,12 @@ public ConfigDO config() {
         configDO.setAuthUser(authUser);
         configDO.setEnv(env);
         configDO.setPersonSearchServiceConfigs(personServiceConfigurations.getConfigs());
+        configDO.setFitsServiceAvailable(getFitsServiceAvailability());
 
         return configDO;
     }
+
+    private Boolean getFitsServiceAvailability() {
+        return fitsUrl.isPresent() && !fitsUrl.toString().isEmpty();
```
Comment (rekt-hard) [inline src/main/java/at/ac/tuwien/damap/rest/ConfigResource.java]: URL performs some checks on the provided string (e.g. protocol, address) and will throw, if it is not a valid URL. Therefore, the second part of the condition can be omitted. What do you think?

File: src/main/java/at/ac/tuwien/damap/rest/config/domain/ConfigDO.java [hunk 2]
```diff
@@ -16,4 +16,5 @@ public class ConfigDO {
     private String authUser;
     private String env;
     private List<ServiceConfig> personSearchServiceConfigs;
+    private Boolean fitsServiceAvailable;
```
Comment (rekt-hard) [inline src/main/java/at/ac/tuwien/damap/rest/config/domain/ConfigDO.java]: I would lean towards using primitive types wherever possible :)

**Review by rekt-hard — COMMENTED** [review]
Nice work. Just minor comments

File: src/main/java/at/ac/tuwien/damap/rest/ConfigResource.java [hunk 3]
```diff
@@ -44,7 +50,12 @@ public ConfigDO config() {
         configDO.setAuthUser(authUser);
         configDO.setEnv(env);
         configDO.setPersonSearchServiceConfigs(personServiceConfigurations.getConfigs());
+        configDO.setFitsServiceAvailable(getFitsServiceAvailability());
 
         return configDO;
     }
+
+    private Boolean getFitsServiceAvailability() {
```
Comment (ValentinFutterer) [inline src/main/java/at/ac/tuwien/damap/rest/ConfigResource.java]: Can you change this one to boolean too :)

**Review by rekt-hard — APPROVED** [review]
Looking good, thank you!

**Review by ValentinFutterer — APPROVED** [review]
Nice!

**Pull request merged**